docs/guides/web/nginx-mainline.md

```diff
@@ -0,0 +1,282 @@
+
+```bash
+sudo firewall-cmd --permanent --zone=public --add-port=443/tcp
+```
+
+These configurations won’t take effect until you force the issue. To do that, tell *firewalld* to relead its configurations, like so:
+
+```bash
+sudo firewall-cmd --reload
+```
+
+Now, there’s a very small chance that this won’t work. In those rare cases, make *firewalld* do your bidding with the old turn-it-off-and-turn-it-on-again.
+
+```bash
+systemctl restart firewalld
+```
+
+To make sure the ports have been added properly, run `firewall-cmd --list-all`. A properly-configured firewall will look a bit like this (I have a few extra ports open on my local server, ignore them):
+
+```bash
+public (active)
+  target: default
+  icmp-block-inversion: no
+  interfaces: enp9s0
+  sources:
+  services: cockpit dhcpv6-client ssh
+  ports: 81/tcp 444/tcp 15151/tcp 80/tcp 443/tcp
+  protocols:
+  forward: no
+  masquerade: no
+  forward-ports:
+  source-ports:
+  icmp-blocks:
+  rich rules:
+```
+
+And that should be everything you need, firewall-wise.
+
+*Now* you should be able to see a web page that looks something like this:
+
+![The Nginx welcome page](nginx/images/welcome-nginx.png)
+
+It’s not much at all, but it means the server is working.
+
+## Creating a Server User and Changing the Website Root Folder
+
+While you *can* just drop your website into the default directory and go (and this might be fine for *Nginx* when it’s running inside a container, or on a test/development server), it’s not what we call best practice. Instead, it’s a good idea to create a specific Linux user on your system for your website, and put your website files in a directory made just for that user. 
+
+If you want to build multiple websites, it’s actually a good idea to create multiple users and root directories, both for the sake of organization and the sake of security. 
+
+In this guide, I’m going to have just the one user: a handsome devil named “www”. We’re going to put all of his website files under a directory in its home folder: `/home/www/`. You can actually put the folder anywhere you want, but on a dedicated server machine, using home folders makes perfect sense.
+
+### Creating the User
+
+First, we make the folder we’re going to use:
+
+```bash
+sudo mkdir /home/www/
+```
+
+Then, we create the user:
+
+```bash
+sudo adduser -g 'Nginx www user' -h /home/www/ www
+```
+
+That command tells the machine to
+
+* Make a user called “www” (as per the last bit of text), 
+* put all of its files in `/home/www`,
+* and add it to the following groups: “Nginx”, “www”, and “user”.
+
+All three are important, but that “Nginx” group does some real magic. It allows the web server to read and modify files that belong to the “www” user, and the “www” user group. See the Rocky Linux [guide to user management](../../books/admin_guide/06-users.md) for more information.
+
+You will, at this point, be prompted to give the new user a password. Type it in, press Enter, and repeat.
```

Comment:
Don't give a password to a system user if you can.

Reply:
See above.